pallets/frag/src/lib.rs

```diff
@@ -0,0 +1,331 @@
+/// the types with this pallet-specific identifier.
+pub mod crypto {
+	use super::KEY_TYPE;
+	use sp_core::ed25519::Signature as Ed25519Signature;
+	use sp_runtime::{
+		app_crypto::{app_crypto, ed25519},
+		traits::Verify,
+		MultiSignature, MultiSigner,
+	};
+	app_crypto!(ed25519, KEY_TYPE);
+
+	pub struct FragAuthId;
+
+	impl frame_system::offchain::AppCrypto<MultiSigner, MultiSignature> for FragAuthId {
+		type RuntimeAppPublic = Public;
+		type GenericSignature = sp_core::ed25519::Signature;
+		type GenericPublic = sp_core::ed25519::Public;
+	}
+
+	// implemented for mock runtime in test
+	impl frame_system::offchain::AppCrypto<<Ed25519Signature as Verify>::Signer, Ed25519Signature>
+		for FragAuthId
+	{
+		type RuntimeAppPublic = Public;
+		type GenericSignature = sp_core::ed25519::Signature;
+		type GenericPublic = sp_core::ed25519::Public;
+	}
+}
+
+use codec::{Compact, Decode, Encode};
+pub use pallet::*;
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+use sp_io::{
+	crypto as Crypto, hashing::blake2_256, hashing::keccak_256, offchain, transaction_index,
+};
+use sp_runtime::offchain::storage::StorageValueRef;
+use sp_std::{
+	collections::{btree_map::BTreeMap, btree_set::BTreeSet},
+	vec,
+	vec::Vec,
+};
+
+use frame_system::offchain::{
+	AppCrypto, CreateSignedTransaction, SendUnsignedTransaction, SignedPayload, Signer,
+	SigningTypes,
+};
+
+pub use weights::WeightInfo;
+
+use sp_clamor::{http_json_post, Hash256};
+
+use scale_info::prelude::{format, string::String};
+
+use serde_json::{json, Value};
+
+#[derive(Encode, Decode, Clone, scale_info::TypeInfo, Debug, PartialEq, Eq)]
+pub struct EthStakeUpdate<TPublic, TBalance> {
+	pub public: TPublic,
+	pub amount: TBalance,
+	pub account: H160,
+}
+
+impl<T: SigningTypes, TBalance: Encode> SignedPayload<T> for EthStakeUpdate<T::Public, TBalance> {
+	fn public(&self) -> T::Public {
+		self.public.clone()
+	}
+}
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::{dispatch::DispatchResult, pallet_prelude::*};
+	use frame_system::pallet_prelude::*;
+	use hex::FromHex;
+	use sp_runtime::{
+		offchain::HttpRequestStatus, traits::AccountIdConversion, MultiSignature,
+		SaturatedConversion,
+	};
+
+	/// Configure the pallet by specifying the parameters and types on which it depends.
+	#[pallet::config]
+	pub trait Config:
+		frame_system::Config + CreateSignedTransaction<Call<Self>> + pallet_assets::Config
+	{
+		/// Because this pallet emits events, it depends on the runtime's definition of an event.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		type WeightInfo: WeightInfo;
+
+		#[pallet::constant]
+		type FragToken: Get<<Self as pallet_assets::Config>::AssetId>;
+
+		#[pallet::constant]
+		type EthChainId: Get<u64>;
+
+		/// The identifier type for an offchain worker.
+		type AuthorityId: AppCrypto<Self::Public, Self::Signature>;
```

Comment:
Explanation to Self: This tell the runtime that this pallet can create signed transactions. 
(https://substrate.recipes/off-chain-workers/transactions.html)